### ISSUE TITLE
Add a way to synch counts and pages between html table rows and queryset counts

### ISSUE THREAD

**Issue by hepcat72** [Princeton-LSI-ResearchComputing/tracebase#388]
### Feature Inspiration

After wrapping up a PR for issue #367, I started thinking about how to deal with representing row counts and handling pagination when the number of rows in the HTML table is noth the same as the queryset.  I happened upon these 2 stack posts:

[explicitly creating the linking tables](https://stackoverflow.com/a/37650905/2057516)
[using the secret `through` reference](https://stackoverflow.com/a/56398474/2057516)

It inspired me to tackle the problem of HTML table row count not matching the queryset count in 1 of 2 ways

### Feature Description

See if there's a way to use the through object to get an accurate html table row count.  POC: My sandbox has one animal that belongs to 2 studies (7 animals total).  Using the `through` object, I can get a queryset that represents the full join:

```
In [14]: Animal.studies.through.objects.count()
Out[14]: 8

In [15]: Animal.objects.count()
Out[15]: 7
```

### Alternatives Considered

Alternatively, we could replace automatically created (and hidden) linking tables with explicitly defined linking tables, as depicted in the [first stack post I referenced above](https://stackoverflow.com/a/37650905/2057516).

### Comment

I'm not sure how using `through` could work with the advanced search interface without fleshing it out entirely with every model instead of doing it all from the root model.

-----

### Assumptions

- List of assumptions made WRT the code
- E.g. We will assume input is correct (explaining why there is no validation)

### Requirements

- List of conditions to be met for the feature
- E.g. Every column/row must display a value, i.e. cannot be empty

### Limitations

- A list of things this work will specifically not do
- E.g. This feature will only handle the most frequent use case X

### Affected/Changed Components

- Files
- Environment variables
- External executables
- Database tables
- Cron job settings
- Etc.

#### GUI Change description

Describe changes the user will see.

#### Code Change Description *(Pseudocode optional)*

Describe code changes planned for the feature.

#### Tests

A test should be planned for each requirement (above), where possible.

**Comment by hepcat72** [comment]
From that second stack post (in a comment on one of the answers), it's mentioned that you may be able to do the following with the `Q` expressions...

If you have 2 models:

- Recipe
- Ingredients

that are related M:M with one another, you can use something like this in the `Q` expression:

```
recipe__ingredients_through_reverse_to_recipe__ingredients_through_reverse_to_ingredients__ingredient_field__iexact=1
```

Not sure about the duplicate `through` inclusion.  I suspect it's maybe a typo, but I'll experiment in the shell today and look in the docs and figure it out.

Played around a bit in the shell and worked out some filter expressions for measured compounds...

```
In [15]: PeakGroup.compounds.through.objects.filter(compound__name__exact="citrate").count()
Out[15]: 106

In [18]: PeakGroup.compounds.through.objects.filter(Q(compound__name__exact="citrate") | Q(compound__name__exact="isocitrate")).count()
Out[18]: 212
```

Note that the rows or "root table" records doubled because they're now "joined" in true join fashion...

**Comment by hepcat72** [comment]
Ran into an unexpected issue with this plan.  When I run the same searches in the shell:
```
PeakGroup.objects.filter(Q(compound__name__exact="citrate")).count()
PeakGroup.objects.filter(Q(compound__name__exact="citrate") | Q(compound__name__exact="isocitrate")).count()
```
...I get the same numbers of results... but when I do the same search on the advanced search page, I get 106 for both queries.  The executions weren't exactly the same, but I didn't think the differences would result in an actual difference, so I am going to see if I can figure out the reason for the difference in the shell.

**Comment by hepcat72** [comment]
Resolved by #401
